### ISSUE TITLE
How can I train the supervised version

### ISSUE THREAD

**Issue by MrZ19** [ZENGYIMING-EAMON/CorrNet3D#3]
Hi, I want to train the supervised version, what should I do? using the Euclidean distance between the index_ and p in line 342, 345? Could you please share your pre-trained models of this version?

**Comment by ZENGYIMING-EAMON** [comment]
Yes, you are right, you can just rewrite the loss calculation function:
def _run_loss(self, txyz1, txyz2, p, out_a, out_b): 
here you can constrain the p with p_label.

**Comment by sakshamjindal** [comment]
One question that I have is that the p matrix that I get from here is of size (B, 1024, 1024). I observe that the if I sum along the columns, each column add up to 1. When I prepare the dataloader, I have only one "1" in each row which means that p_label(i,j) = 1 will hold true for only one element in each row. However, p matrix that I am getting from the _run_step is in transposed form. So while calculating loss, I should be doing something like

`supervised loss = torch.mean(self._batch_frobenius_norm(p, label.transpose(1, 2)))`

What I want to understand regarding correspondence - if p(j, i) = 1, then should I interpret p matrix as having jth point in point cloud 1 correponds to ith point in point cloud 2 (since i am getting p in transposed form) ?

**Comment by sakshamjindal** [comment]
If the above is true, I have some doubts regarding this statement: 

`pred_corr=torch.bmm(p, xyz2.transpose(2, 1).contiguous())`

In the paper it is given that `A_hat = P. B`. Here `B =  xyz2.transpose(2, 1).contiguous())`. Shouldn't it be
 
`pred_corr=torch.bmm(p.transpose(2, 1), xyz2.transpose(2, 1).contiguous())` 

since the p that I am getting from `_run_step()` is transposed,  which means, for each element of the batch, the matrix sums along the column to be 1.  If it were not in tranposed form surely, would have gone ahead with `pred_corr=torch.bmm(p, xyz2.transpose(2, 1).contiguous())` but in this case, shouldn't it be `pred_corr=torch.bmm(p.transpose(2, 1), xyz2.transpose(2, 1).contiguous())`
